prov/efa/src/rxr/rxr_pkt_type_data.c

```diff
@@ -304,16 +297,101 @@ int rxr_pkt_proc_data(struct rxr_ep *ep,
 
 		rxr_msg_multi_recv_free_posted_entry(ep, rx_entry);
 		rxr_release_rx_entry(ep, rx_entry);
-		return 0;
+		return;
 	}
 
 	if (!rx_entry->window) {
 		assert(rx_entry->state == RXR_RX_RECV);
-		ret = rxr_pkt_post_ctrl_or_queue(ep, RXR_RX_ENTRY, rx_entry, RXR_CTS_PKT, 0);
+		err = rxr_pkt_post_ctrl_or_queue(ep, RXR_RX_ENTRY, rx_entry, RXR_CTS_PKT, 0);
+		if (OFI_UNLIKELY(err)) {
+			FI_WARN(&rxr_prov, FI_LOG_CQ, "Cannot post CTS packet\n");
+			rxr_cq_handle_rx_error(ep, rx_entry, err);
+			rxr_release_rx_entry(ep, rx_entry);
+		}
 	}
 
 	rxr_pkt_entry_release_rx(ep, pkt_entry);
-	return ret;
+}
+
+
+/* rxr_pkt_proc_data() processes data of a DATA pkt entry or a READRSP
+ * pkt entry.
+ *
+ * It can be called by either rxr_pkt_handle_data_recv() or
+ * rxr_pkt_handle_readrsp_recv() depend on packet type.
+ *
+ * It will copy data from data pkt_entry to receiving buffer in
+ * rx_entry.
+ *
+ * There can be two cases:
+ *
+ *    1. If the receiving buffer is on host memory, this function will
```

Comment:
I thought you said you fixed this? There's no reason to repeat the two cases on each function that does it, this should be documented in one place.